app.rb

```diff
@@ -0,0 +1,122 @@
+require_relative './book'
+require_relative './classroom'
+require_relative './person'
+require_relative './rental'
+require_relative './student'
+require_relative './teacher'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def booklist
+    puts "\nNo book found!" if @books.empty?
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+  end
+
+  def peoplelist
+    puts "\nNo student or teacher found!." if @people.empty?
+    @people.each { |person| puts "[#{person.class}], Name: #{person.name}, Age: #{person.age}, Id: #{person.id}" }
+  end
+
+  def create_person
+    print "\nDo you want to create a student (1) or a teacher (2)? [Input the number]: "
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts "\nInvalid input! Aborting..."
+    end
+  end
+
+  def create_teacher
+    print "teacher\'s name: "
+    name = gets.chomp
+
+    print "teacher\'s age: "
+    age = gets.chomp
+
+    print "teacher\'s specialization: "
+    specialization = gets.chomp
+
+    teacher = Teacher.new(age, specialization, name, parent_permission: true)
+    puts "\nTeacher created successfully!"
+    @people.push(teacher)
+  end
+
+  def create_student
+    print "student\'s name: "
+    name = gets.chomp
+
+    print "student\'s age: "
+    age = gets.chomp
+
+    print 'has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.upcase
+    case parent_permission
+    when 'N'
+      student = Student.new(name, age, parent_permission: false)
+      @people.push(student)
+    when 'Y'
+      student = Student.new(name, age, parent_permission: true)
+      @people.push(student)
+    end
+    puts "\nStudent created successfully!"
```

Comment:
- [ ] Kindly when I try it gives me an error like in the pic, and that's because here you are creating the student and sending only 3 params `Student.new(name, age, parent_permission: false)` but the student initialize takes 4 params `initialize(name, age, classroom, parent_permission: true)`.Kindly send also the `classroom` when you create the student object.

![image](https://user-images.githubusercontent.com/35535130/180282199-563e30b9-6733-4bdc-b920-94dc5db2e0af.png)

Reply:
It works on my computer, what then can be causing this it to throw this error on yours @keroloussamy